### PR TITLE
add sync editor

### PR DESCRIPTION
**Pull request by jeswanke**
Signed-off-by: John Swanke <jswanke@redhat.com>

### CODE REVIEW

**Comment by KevinFCormier** [comment]
@jeswanke I found a few issues.

1. The **Editor changes** section uses a 14px heading, which is the same as the label on fields. I think this might need to be 20px to match the section headings, and some feedback from design might be useful.
1. Secrets are not hidden in the **Editor changes** section if I reveal them, make changes, then hide them again.
1. Edit credential fails on submit.

image

```
TypeError
Cannot read properties of null (reading 'getAllDecorations')
Call Stack
 getResourceEditorDecorations
  src/components/SyncEditor/decorate.ts:111:10
 decorate
  src/components/SyncEditor/decorate.ts:24:60
 eval
  src/components/SyncEditor/SyncEditor.tsx:246:64
```

image

**Comment by KevinFCormier** [comment]
@jeswanke Thanks for the updates. It's much closer now, except that the secret is not actually updated when I edit. The page does not crash and I get the toast saying it was successful, but it is not actually changed.

File: frontend/src/components/AcmDataForm.tsx [hunk 1]
```diff
@@ -107,11 +113,13 @@ function requiredValidationMessage() {
 
 const minWizardSize = 1000
 const defaultPanelSize = 600
+const EDITOR_CHANGES = 'Editor changes'
```
Comment (KevinFCormier) [inline frontend/src/components/AcmDataForm.tsx:116]: Can we externalize this string, please?

**Review by KevinFCormier — CHANGES_REQUESTED** [review]
@jeswanke I requested you externalize your strings. Also, when YAML editing is off and you go to create/edit/view a Credential, the drawer starts to pop out for a moment before it hides itself. That seems like a bug.

File: frontend/src/components/AcmDataForm.tsx (hunk 1 above)
Comment (jeswanke) [reply frontend/src/components/AcmDataForm.tsx:117]: ok, now t( )

**Pull request closed without merging**